File: 1987-substrings-of-size-three-with-distinct-characters/substrings-of-size-three-with-distinct-characters.java
class Solution {
    public int countGoodSubstrings(String s) {
        int left = 0;
        int count = 0;

        for(int right = 2; right < s.length(); right++)
        {
            char c1 = s.charAt(left);
            char c2 = s.charAt(left + 1);
            char c3 = s.charAt(right);

            if(c1 != c2 && c2 != c3 && c1 != c3) count++;

            left++;
        }
        return count;
    }
}

File: 1987-substrings-of-size-three-with-distinct-characters/substrings-of-size-three-with-distinct-characters.go
func countGoodSubstrings(s string) int {
    count := 0
    window := make(map[byte]int)
    left := 0

    for right := 0; right < len(s); right++ {
        window[s[right]]++

        if right - left + 1 > 3 {
            window[s[left]]--

            if window[s[left]] == 0 {
                delete(window, s[left])
            }

            left++
        }

        if right - left + 1 == 3 && len(window) == 3 {
            count++
        }
    }
    return count
}